### PR TITLE
fix: check to see if prefix is before build and visa versa in isBuild isPr…

### PR DESCRIPTION
**Pull request by yasirfolio3**
## Summary
- Fix isPrerelease and isBuild to check the order of "-" and "+".

# Tests
- Added unit tests.

### CODE REVIEW

File: pkg/decision/evaluator/matchers/semver.go [hunk 1]
```diff
@@ -65,16 +66,12 @@ func (sv SemanticVersion) compareVersion(attribute string) (int, error) {
 			return -1, nil
 		case !sv.isNumber(versionParts[idx]):
 			// Compare strings
-			if versionParts[idx] < targetedVersionParts[idx] {
-				return -1, nil
-			} else if versionParts[idx] > targetedVersionParts[idx] {
-				return 1, nil
+			if result := sv.compareVersionStrings(versionParts[idx], targetedVersionParts[idx], attribute); result != 0 {
```
Comment (msohailhussain) [inline pkg/decision/evaluator/matchers/semver.go:69]: any reason to put it in a separate function?
if we need to, please rename this function name, compareBuildMetadata.

**Review by msohailhussain — APPROVED** [review]
lgtm

File: pkg/decision/evaluator/matchers/semver.go (hunk 1 above)
Comment (yasirfolio3) [reply pkg/decision/evaluator/matchers/semver.go:69]: had to create these functions to resolve cyclo-complexity warning thrown by linter.

File: pkg/decision/evaluator/matchers/semver.go [hunk 2]
```diff
@@ -19,11 +19,12 @@ package matchers
 
 import (
 	"fmt"
-	"github.com/optimizely/go-sdk/pkg/logging"
 	"regexp"
 	"strconv"
 	"strings"
 
+	"github.com/optimizely/go-sdk/pkg/logging"
+
```
Comment (thomaszurkan-optimizely) [inline pkg/decision/evaluator/matchers/semver.go:27]: nit: can we put these in alphabetical order?

**Review by thomaszurkan-optimizely — APPROVED** [review]
LGTM! Thanks!

**Review by pawels-optimizely — APPROVED** [review]
lgtm

**Pull request merged**